src/core/js/views/drawerView.js

```diff
@@ -1,285 +1,288 @@
-      if (!this._isVisible) {
-        Adapt.a11y.popupOpened(this.$el);
-        Adapt.a11y.scrollDisable('body');
-        this._isVisible = true;
-      }
+  toggleDrawer() {
+    if (this._isVisible && this._isCustomViewVisible === false) {
+      this.hideDrawer();
+    } else {
+      this.showDrawer(true);
+    }
+  }
+
+  showDrawer(emptyDrawer) {
+    this.$el.removeClass('u-display-none').removeAttr('aria-hidden');
+    // Only trigger popup:opened if drawer is visible, pass popup manager drawer element
+    if (!this._isVisible) {
+      Adapt.a11y.popupOpened(this.$el);
+      Adapt.a11y.scrollDisable('body');
+      this._isVisible = true;
+    }
 
-      // Sets tab index to 0 for all tabbable elements in Drawer
-      this.$('a, button, input, select, textarea').attr('tabindex', 0);
+    // Sets tab index to 0 for all tabbable elements in Drawer
+    this.$('a, button, input, select, textarea').attr('tabindex', 0);
 
-      if (emptyDrawer) {
-        this.$('.drawer__back').addClass('u-display-none');
+    if (emptyDrawer) {
+      this.$('.drawer__back').addClass('u-display-none');
+      this._isCustomViewVisible = false;
+      this.emptyDrawer();
+      if (this.collection.models.length === 1) {
+        // This callback triggers openCustomView() and sets
+        // _isCustomViewVisible to true, causing toggleDrawer()
+        // to re-render the drawer on every toggle button press
+        Adapt.trigger(this.collection.models[0].get('eventCallback'));
+        // Set _isCustomViewVisible to false to prevent re-rendering
+        // the drawer and fix the toggle functionality on toggle button press
         this._isCustomViewVisible = false;
-        this.emptyDrawer();
-        if (this.collection.models.length === 1) {
-          // This callback triggers openCustomView() and sets
-          // _isCustomViewVisible to true, causing toggleDrawer()
-          // to re-render the drawer on every toggle button press
-          Adapt.trigger(this.collection.models[0].get('eventCallback'));
-          // Set _isCustomViewVisible to false to prevent re-rendering
-          // the drawer and fix the toggle functionality on toggle button press
-          this._isCustomViewVisible = false;
-        } else {
-          this.renderItems();
-          Adapt.trigger('drawer:openedItemView');
-        }
       } else {
-        if (this._hasBackButton && this.collection.models.length > 1) {
-          this.$('.drawer__back').removeClass('u-display-none');
-        } else {
-          this.$('.drawer__back').addClass('u-display-none');
-        }
-        Adapt.trigger('drawer:openedCustomView');
+        this.renderItems();
+        Adapt.trigger('drawer:openedItemView');
+      }
+    } else {
+      if (this._hasBackButton && this.collection.models.length > 1) {
+        this.$('.drawer__back').removeClass('u-display-none');
+      } else {
+        this.$('.drawer__back').addClass('u-display-none');
```

Comment:
could do a toggleClass here `this.$('.drawer__back').toggleClass('u-display-none', this._hasBackButton && this.collection.models.length > 1)`